go/test/endtoend/cluster/vttablet_process.go

```diff
@@ -628,3 +628,69 @@ func VttabletProcessInstance(port, grpcPort, tabletUID int, cell, shard, keyspac
 
 	return vttablet
 }
+
+// SetSuperReadOnly switch-off super-read-only flag in db
+func (vttablet *VttabletProcess) SetSuperReadOnly(dbname string, enableSuperReadOnly bool) error {
+	conn, err := vttablet.defaultConn(dbname)
+	if err != nil {
+		log.Infof("error in getting connection object %s", err)
+		return err
+	}
+	defer conn.Close()
+
+	_, err = conn.SetSuperReadOnly(enableSuperReadOnly)
+	return err
+}
+
+// SetReadOnly switch-off read-only flag in db
+func (vttablet *VttabletProcess) SetReadOnly(dbname string, enableReadOnly bool) error {
+	conn, err := vttablet.defaultConn(dbname)
+	if err != nil {
+		log.Infof("error in getting connection object %s", err)
+		return err
+	}
+	defer conn.Close()
+
+	_, err = conn.SetReadOnly(enableReadOnly)
+	return err
+}
+
+// QueryTabletWithSuperReadOnlyHandling lets you execute a query in this tablet while disabling super-read-only and get the result
+// It will enable super-read-only once its done executing the query.
+func (vttablet *VttabletProcess) QueryTabletWithSuperReadOnlyHandling(query string, keyspace string, useDb bool) (*sqltypes.Result, error) {
+	if !useDb {
+		keyspace = ""
+	}
+	dbParams := NewConnParams(vttablet.DbPort, vttablet.DbPassword, path.Join(vttablet.Directory, "mysql.sock"), keyspace)
+	conn, err := vttablet.conn(&dbParams)
+	if err != nil {
+		return nil, err
+	}
+	defer conn.Close()
+	return executeQueryWithSuperReadOnlyHandling(conn, query)
+}
+
+// executeQuery will retry the query up to 10 times with a small sleep in between each try.
+// This allows the tests to be more robust in the face of transient failures. It disables
+// super-read-only during query execution.
+func executeQueryWithSuperReadOnlyHandling(dbConn *mysql.Conn, query string) (*sqltypes.Result, error) {
+	var (
```

Comment:
Once the above changes are incorporated, then both these functions will no longer be used and can be removed.